packages/prompts/src/limit-options.ts

```diff
@@ -8,37 +9,129 @@ export interface LimitOptionsParams<TOption> extends CommonOptions {
 	maxItems: number | undefined;
 	cursor: number;
 	style: (option: TOption, active: boolean) => string;
+	columnPadding?: number;
+	rowPadding?: number;
 }
 
+const trimLines = (
+	groups: Array<string[]>,
+	initialLineCount: number,
+	startIndex: number,
+	endIndex: number,
+	maxLines: number
+) => {
+	let lineCount = initialLineCount;
+	let removals = 0;
+	for (let i = startIndex; i < endIndex; i++) {
+		const group = groups[i];
+		lineCount = lineCount - group.length;
+		removals++;
+		if (lineCount <= maxLines) {
+			break;
+		}
+	}
+	return { lineCount, removals };
+};
+
 export const limitOptions = <TOption>(params: LimitOptionsParams<TOption>): string[] => {
```

Comment:
this logic is a bit bonkers so if you want it explaining, feel free to catch me on discord